R/fd_inventory.R

```diff
@@ -0,0 +1,74 @@
+# Inventory data
+
+
+#' Return the raw inventory table.
+#'
+#' @details The columns are as follows:
+#' - `Site` (character): Replicate code. Each replicate contains
+#'   multiple plots with different disturbance treatments.
+#' - `SubplotID` (integer): Subplot ID number. These subplot codes are a
+#' concatenation of the plot (\code{\link{fd_plots}}) and
+#' subplot \code{\link{fd_subplots}} codes.
+#' - `Tag` (integer): Tree tag number.
+#' - `Species` (character): Species code from the USDA Plants Database; see
+#' \url{https://plants.sc.egov.usda.gov/java/}.
+#' - `DBH_cm` (numeric): Diameter at breast height (1.37m), cm.
+#' - `Health_status` (character): live (L) or dead (D).
+#' - `Canopy_status` (character): Canopy status: overstory dominant (OD),
+#' overstory submissive (OS), XXX (SA), understory (UN).
+#' - `Date` (date): Inventory entry date.
+#' - `Notes` (character): Notes.
+#' - `Replicate` (character): Replicate code, extracted from `SubplotID`.
+#' - `Plot` (integer): Plot ID number, extracted from `SubplotID`.
+#' - `Subplot` (character): Subplot code, extracted from `SubplotID`.
+#'
+#' @return A `data.frame` or `tibble`. See "Details" for column descriptions.
+#' @export
+#'
+#' @examples
+#' fd_inventory()
+fd_inventory <- function() {
+  inv <- read_csv_file("forte_inventory.csv")
+  inv$Date <- as.Date(inv$Date, format = "%m/%d/%Y")
+
+  # Split the SubplotID column into more useful individual columns
+  inv$Replicate <- substr(inv$SubplotID, 1, 1)
+  inv$Plot <- as.integer(substr(inv$SubplotID, 2, 3))
+  inv$Subplot <- substr(inv$SubplotID, 4, 4)
+
+  # Currently there's a bad entry in the table. Nuke it. Temporary
+  inv$DBH_cm <- as.numeric(inv$DBH_cm)  # temporary, until we fix row 791
```

Comment:
So base R's `read.csv` can accept a `colClasses` argument that, I think, works more or less the same as `readr::read_csv`'s `col_classes`. Down the road, perhaps we should consider using that?

Reply:
Depending on outcome of #11 discussion, agreed.